unicodecsv/__init__.py

```diff
@@ -183,8 +183,11 @@ def __init__(self, csvfile, fieldnames=None, restkey=None, restval=None,
             # Python 2.5 fieldnames workaround. (http://bugs.python.org/issue3436)
             reader = UnicodeReader(csvfile, dialect, encoding=encoding, *args, **kwds)
             self.fieldnames = _stringify_list(reader.next(), reader.encoding)
-        self.unicode_fieldnames = [_unicodify(f, encoding) for f in
-                                   self.fieldnames]
+        if self.fieldnames is not None:
+            self.unicode_fieldnames = [_unicodify(f, encoding) for f in
+                                       self.fieldnames]
+        else:
+            self.fieldnames = None
```

Comment:
I think you probably meant `self.unicode_fieldnames` here?

Sorry for the delay in responding to this.